src/os/bluestore/BlueFS.h

```diff
@@ -363,9 +423,32 @@ class BlueFS {
       buffer_appender.append_zero(len);
     }
 
+    void append(uint64_t value) {
+      uint64_t l0 = get_buffer_length();
+      ceph_assert(l0 + sizeof(value) <= std::numeric_limits<unsigned>::max());
+      buffer_appender.append((const char*)&value, sizeof(value));
+    }
+
+    void append_hole(uint64_t len) {
+      buffer.append_hole(len);
+    }
+
+    void prepend(uint64_t value) {
+      bufferlist tail;
+      tail.append((const char*)&value, sizeof(value));
```

Comment:
IIRC we tend to use arch independent (primarily I'mean  little vs. big endian) encoding for persistent data. E.g. to be able to move data volumes between different nodes. IMO this direct integer mapping violates that requirement.